### DISCUSSION TITLE
How to setup reverse-proxy on apache

### DISCUSSION THREAD

**Discussion by Im-Fran** [requarks/wiki#3636] [Guides]
I was looking how to setup wiki.js on apache for weeks and I want to share this for those that are looking for support. (This little guide is for Ubuntu)

First go to your apache folder (/etc/apache2/) and open the folder "sites-available", then create the file with your subdomain name, for example "wiki.example.com.conf".

Put this code inside the file:
`<VirtualHost *:443>
  	ServerName wiki.example.com
  	ServerAdmin webmaster@example.com
  
  	SSLEngine on
  	SSLCertificateFile /path/to/cert.pem
  	SSLCertificateKeyFile /path/to/key.pem
  
  	ProxyRequests Off
    ProxyPreserveHost On
    ProxyVia Full

    <Proxy *>
        Require all granted
    </Proxy>
    
  	ProxyPass / http://localhost:3000/
  	ProxyPassReverse / http://localhost:3000/

  	Header always set Strict-Transport-Security "max-age=63072000; includeSubdomains; preload"
</VirtualHost>

<VirtualHost *:80>
    ServerName wiki.example.com
    ServerAdmin webmaster@example.com
    
    RewriteEngine On
    RewriteCond %{HTTP_HOST}   ^wiki.example.com$ [NC]
    RewriteRule ^(.*)$         https://wiki.example.com$1 [R=301,L]
</VirtualHost>

`

Then run this command `sudo a2ensite wiki.example.com.conf`
Then restart apache `sudo systemctl restart apache2`
Also you need to go to your dns settings and create an A record pointing to the server IP

**Comment by CraftingGamerTom** [comment]
```
Job for apache2.service failed because the control process exited with error code. See "systemctl status apache2.service" and "journalctl -xe" for details.
```

`RewriteCond %{HTTP_HOST}   ^ wiki.example.com $ [NC]`

Causes `RewriteCond: bad flag delimiters`

**Reply by TidbitSoftware** [reply]
Remove the spaces so that it reads,
`RewriteCond %{HTTP_HOST} ^wiki.example.com$ [NC]`

**Reply by Im-Fran** [reply]
My bad, I'll fix that :D

**Comment by DarkKingKaan** [comment]
Very helpful. Saved me a ton of time. Thank you! :)

**Comment by TechBreakHelp** [comment]
I tried many configurations for three days, also yours, but none is working on my server.
http://wiki.domain.com:3000 works fine, it's reachable
https://wiki.domain.com says "404 not found" (Apache error)

My config.yml (Wiki.js configuration) has SSL option disabled. Could you help me?

**Reply by TechBreakHelp** [reply]
Solved. Apache has created a config file called /etc/apache2/sites-available/wiki.mydomain.com-le-ssl.conf which modifies the configuration. It is used for port 443 (SSL) and in order it became first. I modified it and now it works properly
